backend/config.go

```diff
@@ -151,6 +157,74 @@ func (c *GrafanaCfg) ConcurrentQueryCount() (int, error) {
 	return i, nil
 }
 
+type SQLConfig struct {
+	RowLimit                      int64
+	DefaultMaxOpenConns           int
+	DefaultMaxIdleConns           int
+	DefaultMaxConnLifetimeSeconds int
+}
+
+func (c *GrafanaCfg) SQL() (SQLConfig, error) {
+	// max open connections
+	maxOpenString, ok := c.config[SQLMaxOpenConnsDefault]
+	if !ok {
+		return SQLConfig{}, errors.New("SQLDatasourceMaxOpenConnsDefault not set in config")
+	}
+
+	maxOpen, err := strconv.Atoi(maxOpenString)
+	if err != nil {
+		return SQLConfig{}, errors.New("SQLDatasourceMaxOpenConnsDefault config value is not a valid integer")
+	}
+
+	// max idle connections
+	maxIdleString, ok := c.config[SQLMaxIdleConnsDefault]
+	if !ok {
+		return SQLConfig{}, errors.New("SQLDatasourceMaxIdleConnsDefault not set in config")
+	}
+
+	maxIdle, err := strconv.Atoi(maxIdleString)
+	if err != nil {
+		return SQLConfig{}, errors.New("SQLDatasourceMaxIdleConnsDefault config value is not a valid integer")
+	}
+
+	// max connection lifetime
+	maxLifeString, ok := c.config[SQLMaxConnLifetimeSecondsDefault]
+	if !ok {
+		return SQLConfig{}, errors.New("SQLDatasourceMaxConnLifetimeDefault not set in config")
+	}
+
+	maxLife, err := strconv.Atoi(maxLifeString)
+	if err != nil {
+		return SQLConfig{}, errors.New("SQLDatasourceMaxConnLifetimeDefault config value is not a valid integer")
+	}
+
+	rowLimitString, ok := c.config[SQLRowLimit]
+	if !ok {
+		return SQLConfig{}, errors.New("RowLimit not set in config")
+	}
+
+	rowLimit, err := strconv.ParseInt(rowLimitString, 10, 64)
+	if err != nil {
+		return SQLConfig{}, errors.New("RowLimit in config is not a valid integer")
+	}
+
+	return SQLConfig{
+		RowLimit:                      rowLimit,
+		DefaultMaxOpenConns:           maxOpen,
+		DefaultMaxIdleConns:           maxIdle,
+		DefaultMaxConnLifetimeSeconds: maxLife,
+	}, nil
+}
+
+func (c *GrafanaCfg) LogUserFacingDefaultError() (string, error) {
```

Comment:
```suggestion
func (c *GrafanaCfg) UserFacingDefaultError() (string, error) {
```

I think this makes more sense, no?

Reply:
yeah.. i wanted to have the word `log` at the beginning to kind of "namespace" it, but it's fine, i can remove it.

Reply:
changed.